client/ayon_core/tools/sceneinventory/view.py

```diff
@@ -735,44 +772,71 @@ def _show_version_dialog(self, item_ids, active_repre_id):
-        }
-        repre_info_by_id = self._controller.get_representation_info_items(
-            repre_ids
-        )
+        project_names = set()
+        repre_ids_by_project = collections.defaultdict(set)
+        for container_item in container_items_by_id.values():
+            repre_id = container_item.representation_id
+            project_name = container_item.project_name
+            project_names.add(project_name)
+            repre_ids_by_project[project_name].add(repre_id)
+
+        # active_project_name = None
+        active_repre_info = None
+        repre_info_by_project = {}
+        version_items_by_project = {}
+        for project_name, repre_ids in repre_ids_by_project.items():
+            repres_info = self._controller.get_representation_info_items(
+                project_name, repre_ids
+            )
+            if active_repre_info is None:
+                # active_project_name = project_name
+                active_repre_info = repres_info.get(active_repre_id)
+
+            product_ids = {
+                repre_info.product_id
+                for repre_info in repres_info.values()
+                if repre_info.is_valid
+            }
+            version_items_by_product_id = self._controller.get_version_items(
+                project_name, product_ids
+            )
+
+            repre_info_by_project[project_name] = repres_info
+            version_items_by_project[project_name] = version_items_by_product_id
 
-        product_ids = {
-            repre_info.product_id
-            for repre_info in repre_info_by_id.values()
-        }
-        active_repre_info = repre_info_by_id[active_repre_id]
         active_version_id = active_repre_info.version_id
-        active_product_id = active_repre_info.product_id
-        version_items_by_product_id = self._controller.get_version_items(
-            product_ids
-        )
-        version_items = list(
-            version_items_by_product_id[active_product_id].values()
-        )
-        versions = {version_item.version for version_item in version_items}
-        product_ids_by_version = collections.defaultdict(set)
-        for version_items_by_id in version_items_by_product_id.values():
-            for version_item in version_items_by_id.values():
+        # active_product_id = active_repre_info.product_id
+
+        versions = set()
+        product_ids = set()
+        version_items = []
+        product_ids_by_version_by_project = {}
+        for project_name, version_items_by_product_id in (
+            version_items_by_project.items()
+        ):
+            product_ids_by_version = collections.defaultdict(set)
+            product_ids_by_version_by_project[project_name] = (
+                product_ids_by_version
+            )
+            versions |= {
+                version_item.version
+                for version_item in version_items_by_product_id.values()
+            }
```

Comment:
do we need to have this loop? 
I think we don't need it because we are going to loop through it in the next for loop.